pkg/controllers/certificatestore_controller.go

```diff
@@ -0,0 +1,155 @@
+//+kubebuilder:rbac:groups=config.ratify.deislabs.io,resources=certificatestores/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// the CertificateStore object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
+func (r *CertificateStoreReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := logrus.WithContext(ctx)
+
+	var resource = req.Name
+	var certStore configv1alpha1.CertificateStore
+
+	logger.Infof("reconciling certificate store '%v'", resource)
+
+	if err := r.Get(ctx, req.NamespacedName, &certStore); err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Infof("deletion detected, removing certificate store %v", req.Name)
+			delete(certificatesMap, resource)
+		} else {
+			logger.Error(err, "unable to fetch certificate store")
+		}
+
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// get cert provider attributes
+	attributes, err := getCertStoreConfig(certStore.Spec)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// fetch certificates based on the provider
+	switch certStore.Spec.Provider {
+	case "azurekeyvault":
+		contents, err := azurekeyvault.GetCertificates(ctx, attributes)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("Error fetching certificates in store %v with azure key vault provider, error: %v", resource, err)
+		}
+
+		certificates, err := byteToCerts(contents)
+		if err != nil {
+			logrus.Error(err)
+			return ctrl.Result{}, fmt.Errorf("Failed to parse x509 certificate for certificate store %v, error: %v", resource, err)
+		}
+		certificatesMap[resource] = certificates
+		logger.Infof("%v certificates fetched for certificate store %v", len(certificates), resource)
+	default:
+		//logger.Errorf("Unknown cert provider %s", certStore.Spec.Provider)
+		return ctrl.Result{}, fmt.Errorf("Unknown provider defined in certificate store %v", resource)
+	}
+
+	// returning empty result and no error to indicate we’ve successfully reconciled this object
+	return ctrl.Result{}, nil
+}
+
+// returns the internal certificate map
+func GetCertificatesMap() map[string][]*x509.Certificate {
+	return certificatesMap
+}
+
+// converts array of cert content to x509 certificate
+func byteToCerts(certificates []types.Certificate) ([]*x509.Certificate, error) {
+	r := []*x509.Certificate{}
+	for _, c := range certificates {
+		block, _ := pem.Decode(c.Content)
+		if block == nil {
+			return nil, fmt.Errorf("Failed to decode certificate")
+		}
+
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		common.LogDebug("cert '%v' added", c.CertificateName)
```

Comment:
Not related to this PR: Why do we not use logrus.Debugf directly? the log level setting for the library by default will selectively print out the debug logs only if level is properly set.

Reply:
reverted.thanks!